lib/llhttp/parser.js

```diff
@@ -68,13 +68,15 @@ const wasm_on_headers_complete = p => {
   let url
   let statusCode
   let statusMessage
-  const upgrade = inst.exports.llhttp_get_upgrade(p)
+  const upgrade = Boolean(inst.exports.llhttp_get_upgrade(p))
   const shouldKeepAlive = inst.exports.llhttp_should_keep_alive(p)
 
   for (let c = 0; c < i[kHeadersFields].length; c++) {
     rawHeaders.push(i[kHeadersFields][c], i[kHeadersValues][c])
   }
-  i[kOnHeaders](rawHeaders)
+  // TODO: understand better when this is called in node parser
+  // and if it's needed. For now it causes duplicated header values
+  // i[kOnHeaders](rawHeaders)
```

Comment:
It’s when the buffer for headers is overflowing. I don’t think it’s needed.

Reply:
Thanks